builder/googlecompute/step_create_instance.go

```diff
@@ -45,6 +44,14 @@ func (c *Config) createInstanceMetadata(sourceImage *Image, sshPublicKey string)
 	} else if wrappedStartupScript, exists := instanceMetadata[StartupScriptKey]; exists {
 		instanceMetadata[StartupWrappedScriptKey] = wrappedStartupScript
 	}
+
+	// Read metadata from files specified with metadata_files
+	for key, value := range c.MetadataFiles {
+		var content []byte
+		content, err = ioutil.ReadFile(value)
```

Comment:
Nice one, I think it would be better to handle the error here, other than that, LGTM

Reply:
🤔 some tests would be welcome as well 🙂

Reply:
@azr I have added the error handling for the method similar to what I found [here](https://github.com/hashicorp/packer/blob/master/builder/googlecompute/step_create_instance.go#L84). Also, I saw that there was no error handling for [reading startup script](https://github.com/hashicorp/packer/blob/master/builder/googlecompute/step_create_instance.go#L43). I have added similar error handling for it too. 

Should I remove the error from return statement, as the error is handled in the `createInstanceMetadata` method and anyway was not handled [here](https://github.com/hashicorp/packer/blob/master/builder/googlecompute/step_create_instance.go#L101)

Reply:
I think it would be better to simply return the error here, ( we could also wrap it for a better context in case 🙂 )

Reply:
Yes good catch, it would be also nice to fail in case the startup script read fails, as this would be otherwise silent.

Reply:
I think it would be better to simply handle error in `Run`

Reply:
@azr If we return a single error, we will have error message for the last metadata file(last entry in the map) only. All previous error will be remain silent. I think we should put all the errors in state or atleast display them on the UI, if possible.

Reply:
Yes I agree that would be nice, we should probably use a [MultiError](https://github.com/hashicorp/packer/blob/098764d8ac1eacfdac362d4e3e1c84b2a504a272/packer/multi_error.go#L11) ! 🙂

Reply:
@azr Done! Please take a look 🙂